packages/grid/x-grid/src/tests/editRows.XGrid.test.tsx

```diff
@@ -254,4 +255,33 @@ describe('<XGrid /> - Edit Rows', () => {
     fireEvent.click(input0);
     expect(document.activeElement).to.have.property('value', 'Nike');
   });
+
+  it('should apply the valueParser before saving the value', () => {
+    const valueParser = stub().withArgs('62').returns(1962);
+    render(
+      <TestCase
+        columns={[
+          { field: 'brand', editable: true },
+          { field: 'year', editable: true, valueParser },
+        ]}
+      />,
+    );
+    const cell = getCell(1, 1);
+    cell.focus();
+    fireEvent.doubleClick(cell);
+    const input = cell.querySelector('input')!;
+    expect(input.value).to.equal('1961');
+
+    fireEvent.change(input, { target: { value: '62' } });
+    expect(cell.querySelector('input')!.value).to.equal('62');
+
+    fireEvent.keyDown(input, { key: 'Enter' });
+    expect(cell).not.to.have.class('MuiDataGrid-cellEditing');
+    expect(cell).to.have.text('1962');
+    expect(valueParser.callCount).to.equal(1);
+    expect(valueParser.args[0][0]).to.equal('62');
+    expect(valueParser.args[0][1].id).to.equal(1);
+    expect(valueParser.args[0][1].field).to.equal('year');
+    expect(valueParser.args[0][1].value).to.equal(1961);
```

Comment:
If it's possible using a `.deep.equal` or `.include`, we might want to use it to make the test case easier to read (self declaration of the JS structure).